hedyorg/hedy
# [BACKEND] Hash parsed programs to prevent duplicate parsing programs

**Description**
In this PR we hash programs after parsing and use a look-up table to check if the program is already parsed previously. This reduces a lot of duplicate server load by parsing identical programs. We would have to keep track of the server to make sure the memory doesn't overflow by this caching, otherwise a maximum size might be nice.

**Fixes**
No related issue.

**How to test**
Run several programs, add some debug prints to `PARSED_PROGRAMS` to verify that only new programs are added to the dictionary.
